src/utils/MainApi.js

```diff
@@ -0,0 +1,85 @@
+export class MainApi {
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+    }
+  
+    _respond(res) {
+        return res.ok
+          ? res.json()
+          : Promise.reject(`Something went wrong: ${res.status}`)
+      ;
+    }
+  
+    // getUserInfo() {
+    //   return fetch(`${this._baseUrl}/users/me`, {
+    //     headers: {
+    //       "Content-Type": "application/json",
+    //       authorization: `Bearer ${localStorage.getItem("jwt")}`,
+    //     }
+    //   }).then(this._respond);
+    // }
+  
+    getSavedArticles(token) {
+      return fetch(`${this._baseUrl}/articles`, {
+        headers: {
+          "Content-Type": "application/json",
+          authorization: `Bearer ${token}`,
+        }
+      }).then(this._respond);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.